include/seqan3/alphabet/all.hpp

```diff
@@ -39,9 +39,127 @@
+ * with seqan3::dna15 requires two bytes per letter (15 * 42 > 256).
+ *
+ * ### assigning and retrieving values
+ *
+ * As mentioned above, we typically think of alphabets in their character representation, but we also
+ * require them in "rank representation" as programmers. In C and C++ it is quite difficult to
+ * cleanly differentiate between these, because the `char` type is considered an integral type and can
+ * be used to index an array (e.g. <tt>my_array['A']</tt> translates to `my_array[65]`). Moreover the sign of `char` is
+ * implementation defined and on many platforms the smallest integer types `int8_t` and `uint8_t` are literally the
+ * same types as `signed char` and `unsigned char` respectively.
+ *
+ * This leads to ambiguity when assigning and retrieving values:
+ *
+ * ```cpp
+ * // WRONG EXAMPLE:
+ * dna4 my_letter{0};      // we want to set the default, an A
+ * dna4 my_letter{'A'};    // we also want to set an A, but we are setting value 65
+ *
+ * std::cout << my_letter; // you expect 'A', but how would you access the number?
+ * ```
+ *
+ * To solve this problem, every alphabet defines two interfaces:
+ *
+ *   1. a **character based interface** with
+ *     1. the underlying character type able to represent this alphabet visually (almost always `char`,
+ * but could be `char16_t` or `char32_t`, as well)
+ *     2. a seqan3::to_char function to produce the visual representation
+ *     3. a seqan3::assign_char function to assign from the visual representation
+ *   2. a **rank based interface** with
+ *     1. the underlying rank type able to represent this alphabet numerically; this type must be able to represent
+ * the numbers from `0` to `alphabet size - 1` (often `uint8_t`, but sometimes a larger unsigned integral type)
+ *     2. a seqan3::to_rank function to produce the numerical representation
+ *     3. a seqan3::assign_rank function to assign from the numerical representation
+ *
+ * To prevent the aforementioned ambiguity, you can neither assign from rank or char representation via `operator=`,
+ * nor can you cast the alphabet to either of it's representation forms, **you need to explicitly use the
+ * interfaces**:
+ * ```cpp
+ * dna4 my_letter;
+ * assign_rank(my_letter, 0);       // assign an A via rank interface
+ * assign_char(my_letter, 'A');     // assign an A via char interface
+ * my_letter = dna4::A;             // some alphabets(BUT NOT ALL!) also provide an enum-like interface
+ *
+ * std::cout << to_char(my_letter);            // prints 'A'
+ * std::cout << (unsigned)to_rank(my_letter);  // prints 0
+ * // we have to add the cast here, because uint8_t is also treated as a char type by default :(
+ * ```
+ *
+ * For efficiency the representation saved internally is normally the rank representation, and the character
+ * representation
+ * is generated via conversion tables, however this is not required as long as both interfaces are provided
```

Comment:
missing comma after however?

Reply:
I am not sure, but I just split it into two sentences.